addons/im_livechat/models/discuss_channel.py

```diff
@@ -241,3 +241,11 @@ def _chatbot_restart(self, chatbot_script):
 
     def _types_allowing_seen_infos(self):
         return super()._types_allowing_seen_infos() + ["livechat"]
+
+    def _filter_for_init_messaging(self):
+        channels = super()._filter_for_init_messaging()
+        if self.env.context.get("is_for_livechat"):
+            # Only returns the latest live chat, every other channel is
+            # irrelevant in the live chat context.
+            return channels.filtered(lambda c: c.channel_type == 'livechat' and c.livechat_active)[-1:]
+        return channels
```

Comment:
This is more LOCs and processing, but I don't mind as conceptually it goes in the right direction to reduce knowledge in client code for what we need.

Just wondering whether returning all livechat introduce bugs? If so, probably the JS code relies too much on the whole state of the code and it doesn't combine well with the rest of Discuss code. Usually when this happens, this means that there's too much opinionated custom code in livechat and/or lack of data in formatters.

Ok for now, but if there's this issue that I mentioned in previous paragraph, we should continue with further code cleaning in livechat. I'm confident this will result in less custom code in livechat, so the effort is worth the price :)